### PR TITLE
feat: peer to peer routing

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

File: backend/controller/encryption/integration_test.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ func WithEncryption() in.Option {
 }
 
 func TestEncryptionForLogs(t *testing.T) {
+	t.Skip("This test needs the timeline service refactoring done")
```
Comment (alecthomas) [inline backend/controller/encryption/integration_test.go:34]: Encryption will be reimplemented, so you can probably jut delete this completely.

**Review by alecthomas — APPROVED** [review]

File: internal/deploymentcontext/module_context.go [hunk 2]
```diff
@@ -101,6 +103,14 @@ func (b *Builder) AddSecrets(secrets map[string][]byte) *Builder {
 	return b
 }
 
+// AddRoutes adds configuration values (as bytes) to the builder
```
Comment (alecthomas) [inline internal/deploymentcontext/module_context.go:106]: This comment makes no sense.

File: backend/cron/service.go [hunk 3]
```diff
@@ -46,7 +43,7 @@ func (c cronJob) String() string {
 }
 
 // Start the cron service. Blocks until the context is cancelled.
-func Start(ctx context.Context, eventSource schemaeventsource.EventSource, verbClient CallClient) error {
+func Start(ctx context.Context, eventSource schemaeventsource.EventSource, client routing.CallClient) error {
```
[the remark this answers is not in the document]
Comment (stuartwdouglas) [reply backend/cron/service.go:46]: Ah yea, I already have that to allow the controller and provisioner to allow access. I will update it to allow ingress and cron for modules that need those services.

File: internal/deploymentcontext/module_context.go (hunk 2 above)
Comment (stuartwdouglas) [reply internal/deploymentcontext/module_context.go:106]: copypasta!

**Comment by stuartwdouglas** [comment]
I am going to disable the rolling deployment part of the kube tests, I don't think there is any point in worrying about that until well after the refactor is done.

**Pull request merged**